sherpa/astro/io/__init__.py

```diff
@@ -1147,7 +1155,25 @@ def _pack_arf(dataset):
     return data, header
 
 
-def _make_int_array(rows, ncols):
+def _find_int_dtype(rows: Sequence[Sequence[int]]) -> type:
+    """What data type should represent the matrix of integers?
+
+    There is no guarantee that each row has the same number of
+    elements.
+
+    """
+
+    maxval = 0
+    for row in rows:
+        if len(row) == 0:
+            continue
+
+        maxval = max(maxval, np.max(row))
+
+    return np.int32 if maxval > 32767 else np.int16
+
```

Comment:
I know this is not in the critical path, but why not return early?
```suggestion
    for row in rows:
        if len(row) == 0:
            continue
        if np.max(row) > 32767:
            return np.int32
    return np.int16

```